include/alpaka/atomic/AtomicStlLock.hpp

```diff
@@ -46,6 +51,34 @@ namespace alpaka
                 typename TSfinae>
             friend struct atomic::traits::AtomicOp;
 
+            static constexpr size_t nextPowerOf2(size_t const value, size_t const bit = 0)
+            {
+                return value <= (1u << bit) ? (1u << bit) : nextPowerOf2(value, bit + 1u);
```

Comment:
Visual Studio 2017 says: 
```
C:\projects\alpaka\include\alpaka/atomic/AtomicStlLock.hpp(56): warning C4334: '<<': result of 32-bit shift implicitly converted to 64 bits (was 64-bit shift intended?)
```
I think 1u is interpreted as 32 bit. Maybe a `static_cast<size_t>(1u)` might help?